tidy3d/plugins/mode/mode_solver.py

```diff
@@ -1141,6 +1143,324 @@ def plot_field(
+        **kwargs,
+    ) -> Ax:
+        """Plot the mode plane cell boundaries as lines.
+
+        Parameters
+        ----------
+        ax : matplotlib.axes._subplots.Axes = None
+            Matplotlib axes to plot on, if not specified, one is created.
+        **kwargs
+            Optional keyword arguments passed to the matplotlib ``LineCollection``.
+            For details on accepted values, refer to
+            `Matplotlib's documentation <https://tinyurl.com/2p97z4cn>`_.
+
+        Returns
+        -------
+        matplotlib.axes._subplots.Axes
+            The supplied or created matplotlib axes.
+        """
+
+        # Get the mode plane normal axis, center, and limits.
+        axes = [0, 1, 2]
+        a_center = [None, None, None]
+        n_axis = self.normal_axis
+        axes.pop(n_axis)
+        a_center[n_axis] = self.plane.center[n_axis]
+        h_lim = [
+            a_center[n_axis] - self.plane.size[axes[0]] / 2,
+            a_center[n_axis] + self.plane.size[axes[0]] / 2,
+        ]
+        v_lim = [
+            a_center[n_axis] - self.plane.size[axes[1]] / 2,
+            a_center[n_axis] + self.plane.size[axes[1]] / 2,
+        ]
+
+        return self.simulation.plot_grid(
+            x=a_center[0], y=a_center[1], z=a_center[2], hlim=h_lim, vlim=v_lim, ax=ax, **kwargs
+        )
+
+    def plot_pml(
+        self,
+        ax: Ax = None,
+    ) -> Ax:
+        """Plot the mode plane absorbing boundaries.
+
+        Parameters
+        ----------
+        ax : matplotlib.axes._subplots.Axes = None
+            Matplotlib axes to plot on, if not specified, one is created.
+
+        Returns
+        -------
+        matplotlib.axes._subplots.Axes
+            The supplied or created matplotlib axes.
+        """
+
+        # Get the mode plane normal axis, center, and limits.
+        axes = [0, 1, 2]
+        coord = ["x", "y", "z"]
+        a_center = [None, None, None]
+        n_axis = self.normal_axis
+        axes.pop(n_axis)
+        coord.pop(n_axis)
+        a_center[n_axis] = self.plane.center[n_axis]
+        h_lim = [
+            a_center[n_axis] - self.plane.size[axes[0]] / 2,
+            a_center[n_axis] + self.plane.size[axes[0]] / 2,
+        ]
+        v_lim = [
+            a_center[n_axis] - self.plane.size[axes[1]] / 2,
+            a_center[n_axis] + self.plane.size[axes[1]] / 2,
+        ]
+
+        # Plot the mode plane is ax=None.
+        if not ax:
+            ax = self.simulation.plot(
+                x=a_center[0],
+                y=a_center[1],
+                z=a_center[2],
+                hlim=h_lim,
+                vlim=v_lim,
+                source_alpha=0,
+                monitor_alpha=0,
+                ax=ax,
+            )
+
+        # Mode plane grid.
+        plane_grid = self.grid_snapped.centers
+        coord_0 = getattr(plane_grid, coord[0])[1:-1]
+        coord_1 = getattr(plane_grid, coord[1])[1:-1]
+
+        # Number of PML layers in ModeSpec.
```

Comment:
there is a `Simulation.pml_thicknesses` method. Could we use this?  or the `Simulation.plot_pml` to avoid duplicating work?

Reply:
The `Simulation.pml_thicknesses` stores only the FDTD boundaries. The mode plane PML thickness is defined by `num_pml` in `ModeSpec`, which actually defines only the number of cells in each plane direction.

Reply:
I think one simple solution would be to make an `updated_copy()` of the simulation, and add some PML layers (just for plotting) corresponding to `num_pml`.  Then you should be able to plot PMLs (and the other methods) normally. Does that make sense / does it seem like it should work?

Reply:
It should work for mode planes spanning into the simulation domain boundaries, but it will fail for mode planes smaller than the simulation domain, right?

Reply:
might be possible to use this method, or the `subsection` method
https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/plugins/mode/mode_solver.py#L1162-L1205

Reply:
And @momchil-flex commented that for the mode solver the PML is included within the mode plane limits.